analytic_engine/Cargo.toml

```diff
@@ -56,4 +56,5 @@ common_util = { workspace = true, features = ["test"] }
 env_logger = { workspace = true }
 pin-project-lite = { workspace = true }
 rand = { workspace = true }
+tempfile = { workspace = true }
```

Comment:
By put `tempfile` here, we can test `analytic_engine` alone, otherwise it will throw errors when `cargo test` in analytic_engine dir:
```
error[E0433]: failed to resolve: use of undeclared crate or module `tempfile`
   --> analytic_engine/src/tests/util.rs:557:19
    |
557 |         let dir = tempfile::tempdir().unwrap();
    |                   ^^^^^^^^ use of undeclared crate or module `tempfile`
```